clang/include/clang/AST/Type.h

```diff
@@ -2746,8 +2751,12 @@ class BuiltinType : public Type {
     return getKind() >= Bool && getKind() <= UInt128;
   }
 
+  bool isDecimalFloatingPoint() const {
+    return getKind() >= DecimalFloat32 && getKind() <= DecimalFloat128;
+  }
+
   bool isFloatingPoint() const {
-    return getKind() >= Half && getKind() <= Ibm128;
+    return getKind() >= Half && getKind() <= DecimalFloat128;
```

Comment:
As discussed offline, I found codegen and lowering much simpler if we treated this like fixed point and acted as if it was a signed integer type. We may want to revisit this once we get to those tasks.

Reply:
Oh, neat, this comment survived the branch rebasing! Maybe all is not lost with respect to stacking and rebasing PRs!

I agree we may want to revisit this, but I think we should try to be consistent with the standard until we have a good indication that doing so isn't working in practice. It might make sense to introduce an `isBinaryFloatingPoint()` predicate and switch most existing uses of `isFloatingPoint()` (~330 uses) to it. Likewise for `isFloatingType` (~85 uses). I hope there won't be incentive to do similarly for `isRealType()` (~12 uses), `isRealFloatingType()` (~100 uses), and `isArithmeticType()` (~40 uses).